custom/enikshay/management/commands/multiple_open_referrals_reconciliation.py

```diff
@@ -74,32 +77,37 @@ def get_case_to_be_retained(self, referral_cases, occurrence_case_id):
 
     def public_app_case(self, occurrence_case_id):
         try:
-            person_case = get_person_case_from_occurrence(DOMAIN, occurrence_case_id)
+            self.person_case = get_person_case_from_occurrence(DOMAIN, occurrence_case_id)
```

Comment:
If the call to `get_person_case_from_occurrence` here fails, then `self.person_case` will still have the value from the previous call, and `get_case_to_be_retained` (which references it) won't know the difference.

it's a bit opaque to set this property in this method and then reference it in another method, especially since this method is called a number of times with different arguments, since it's not clear where that came from.  Another approach would be to fetch it first higher up (in `handle` in this case) and then pass it in to each of these methods.

Reply:
agreed that this looks a little delicate but the way the command is written if `get_person_case_from_occurrence` fails and returns False, there is nothing further done about this case. Going to make this a little more secure by adding `self.person_case=None` before this.

> Another approach would be to fetch it first higher up (in handle in this case) and then pass it in to each of these methods.
passing this around was exactly what i was trying to avoid by assigning this here. :)